OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -318,7 +330,7 @@ void ProtectOwn(IBot bot, Actor attacker)
 			{
 				var ownUnits = World.FindActorsInCircle(World.Map.CenterOfCell(GetRandomBaseCenter()), WDist.FromCells(Info.ProtectUnitScanRadius))
 					.Where(unit => unit.Owner == Player && !unit.Info.HasTraitInfo<BuildingInfo>() && !unit.Info.HasTraitInfo<HarvesterInfo>()
-						&& unit.Info.HasTraitInfo<AttackBaseInfo>());
+						&& !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name) && unit.Info.HasTraitInfo<AttackBaseInfo>());
```

Comment:
We have several places (`FindNewUnits`, `CreateAttackForce`, `TryToRushAttack`, `ProtectOwn`) that all use slightly different hardcoded checks to determine unit validity and type.

Mixing hardcoded aircraft checks with yaml-defined naval checks is really ugly, can we please define `AirUnitsTypes` on `SquadManagerBotModuleInfo` to replace the `AircraftInfo` checks?

`ExcludeFromSquadsTypes` is only used in `FindNewUnits`, with the other places using hardcoded checks to exclude harvesters and non-combat units. Can we please define and use `ExcludeFromSquadsTypes` properly?

Sorry for the scope creep, but if we're going to risk touching this now I really think we should do it properly. If you'd rather not then I guess I can try and summon the motivation to do this.

Reply:
OK... if you feel like trying go ahead, but if there are any signs of this turning into a can of worms, I'd suggest to not bother (at least for the hotfix release).

Reply:
5f040597dbb4183d155adc8bcbc583545d159044 shows what I have in mind on the trait side. We'll need to define the actor lists for each mod now.

Reply:
Implementing this properly is too much for the hotfix, so lets drop this for now.